tests/unit/sagemaker/image_uris/test_algos.py

```diff
@@ -176,21 +177,6 @@ def _accounts_for_algo(algo):
 @pytest.mark.parametrize("algo", ALGO_NAMES)
 def test_algo_uris(algo):
     accounts = _accounts_for_algo(algo)
-
-    for region in regions.regions():
+    for region in accounts:
         uri = image_uris.retrieve(algo, region)
         assert expected_uris.algo_uri(algo, accounts[region], region) == uri
-
-
-def test_lda():
-    algo = "lda"
-    accounts = _accounts_for_algo(algo)
-
-    for region in regions.regions():
-        if region in accounts:
-            uri = image_uris.retrieve(algo, region)
-            assert expected_uris.algo_uri(algo, accounts[region], region) == uri
-        else:
-            with pytest.raises(ValueError) as e:
```

Comment:
Removing these if/else blocks because 1. The if is really filtering the regions and can be combined with the loop above, and since we are removing use of boto regions anyways this makes no sense.

The ValueError raised in else is not something that should be done in a test, especially if we are checking only supported regions, and seems redundant.